api-report/container-runtime.api.md

```diff
@@ -127,7 +127,7 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
     // (undocumented)
     getQuorum(): IQuorumClients;
     // (undocumented)
-    getRootDataStore(id: string, wait?: boolean): Promise<IFluidRouter>;
+    getRootDataStore(id: string, wait?: boolean): Promise<IFluidDataStoreRuntimeEntrypoint>;
```

Comment:
Why root data store? I do not see anything special about it - I think you want to party on IDataStore interface and add to it (and this call likely needs to return same interface)

Reply:
see this comment: https://github.com/microsoft/FluidFramework/pull/11887#discussion_r966540316

Reply:
I did something very close to that first, but @andre4i pointed out [here](https://github.com/microsoft/FluidFramework/pull/11887#discussion_r967326385) that the motivation for `IDataStore` is so its single method isn't present on root data stores, only non-root ones. And we also want the `handle` from `IFluidDataStoreRuntimeEntryPoint` in `IFluidDataStoreChannel` and `IFluidDataStoreRuntime`, which are sometimes used on their own, independently of `IDataStore`.

That said, I still feel so-so about my big-picture understanding of some things around the change, so I'm all ears if there's a better design here.

Reply:
I agree with this in principle, but having a method on an interface which will fail (sort of, if the alias is different) all the time for a specific type of objects is not ideal from the customer's view of the API surface. It's like you're getting an already aliased datastore all the time and you're _not supposed_ to call the trySetAlias method.

Reply:
It will only fail if one uses different alias name. It's same for any other data store (including trying to concurrently give it two different names).
If we really want to avoid it, I'd rather have one interface, but use Omit<> for root data store. I.e. not create new interface names - it sort of makes sense for us, but it's confusing for consumers, especially when we refer to one name in documentation, but they deal with derived interfaces (it's not trivial these are the same things without looking deeper).

Reply:
why did we even create IDataStore and the DataStore class if we want this everywhere. i'd propose we remove those layers and just push aliasing to the channel interface, rather than cement this layer

Reply:
regardless i think whatever happens with trySetAlias is orthogonal to the work happening here, and i don't want to solve those issues here. we can make this change must less invasive to the surface area for now.

Reply:
Channel interface is only for communication to container runtime. Data store authors should never use it for anything else (including returning it to users of data store), same applies to us. It's sort of "internal" thing (in a sense that it should be used only by those who have to implement that interface).

Reply:
I do not believe we expose IFluidDataStoreChannel anywhere, other than when interfacing with IFluidDataStoreContext. If you see otherwise, please point me to it.

IDataStore is implemented only in one place and it definitely does not expose channel:

```
class DataStore implements IDataStore {
```

Reply:
currently:
containerRuntime -> getRootDataStore -> getRootDataStoreChannel

before aliases:
containerRuntime -> _createDataStoreWithProps -> _createDataStore
containerRuntime -> createDataStore -> _createDataStore

This was split during aliasing. Without making any judgements on what the right path forward, i don't want to couple any changes with this change, and it will bring new and untested functionality to our public apis. I think opening a distinct issue to rationalize, and test whatever approach is the right path forward.

Reply:
_createDataStore() is not a public API. While IFluidDataStoreChannel is public interface, it's used only to communicate with context. Data store authors should never expose it, neither should we (outside of context - runtime communication).
So I'm not sure how it's related to a public API data store consumers are interfacing with - such developers should never know about IFluidDataStoreChannel interface.

Reply:
i agree with everything you state, and i have opinions about how it could be made better, but i don't think this pr is the right place to modify it, as it is not really related